comp/core/workloadmeta/store.go

```diff
@@ -450,6 +450,41 @@ func (w *workloadmeta) Reset(newEntities []Entity, source Source) {
 	w.Notify(events)
 }
 
+func (w *workloadmeta) validatePushEvents(events []PushEvent) error {
+	for _, event := range events {
+		if event.Type != EventTypeSet && event.Type != EventTypeUnset {
+			return fmt.Errorf("unsupported PushEvent type: only Set and Unset types are allowed for push events")
+		}
+	}
+	return nil
+}
+
+func (w *workloadmeta) transformPushEvents(events []PushEvent) []CollectorEvent {
+	transformedPushEvents := make([]CollectorEvent, len(events))
+
+	for index, event := range events {
+		transformedPushEvents[index] = CollectorEvent{
+			Type:   event.Type,
+			Source: Source(fmt.Sprintf("external-%s", event.Source)),
```

Comment:
I think I'd need to see a use case to know what's the best option here.
Are we going to have something like `external-runtime` or `external-node_orchestrator` ? Or should we just define a new source called `External` ?

Reply:
Depending on what we decide we might not need to transform anything. Just validate that event.Source == External (or whatever we decide).

Reply:
It might be even better to create a specific source for each use case so that it's possible to subscribe to entities created by that specific source.

Reply:
Defining a new source called `External` is a good idea, except that in that case we won't be able to distinguish events pushed by different external sources. 

For example, if the language detection component pushes an event to set a deployment entity, and then the process component pushes an event to set the same deployment entity, we need to a way to know that these 2 pushed events are coming from different external sources, and we also need to identify each external source (mainly for debugging purposes and to be able to emit telemetry with proper tags).

Reply:
I updated the code to use the `Event` type instead, and use a common `Source` as a separate argument. This solves the issue of preserving the abstraction of `Collectors` without needing to define a new type.

Reply:
Please check `Possible Drawbacks / Trade-offs` in the PR description.